components/add-set-form.tsx

```diff
@@ -0,0 +1,421 @@
+                    </SelectItem>
+                  );
+                })}
+              </SelectGroup>
+            </SelectContent>
+          </Select>
+          <Select
+            onValueChange={(newValue) =>
+              setSelectedVisualizerType(newValue as VisualizerType)
+            }
+          >
+            <SelectTrigger className="w-48">
+              <SelectValue placeholder="Visualizer Type" />
+            </SelectTrigger>
+            <SelectContent>
+              <SelectGroup>
+                <SelectLabel>Visualizer Type</SelectLabel>
+                {visualizerTypes.map((visualizerType) => {
+                  return (
+                    <SelectItem key={visualizerType} value={visualizerType}>
+                      {visualizerType}
+                    </SelectItem>
+                  );
+                })}
+              </SelectGroup>
+            </SelectContent>
+          </Select>
+          <Input
+            type="text"
+            id="setupAlg"
+            placeholder="Setup Alg"
+            className="w-48"
+            onChange={(e) => setSetupAlg(e.target.value)}
+          />
+        </div>
+      </div>
+      <div className="flex flex-col gap-2">
+        <h3>Mask Settings</h3>
+        <div className="flex items-center gap-2">
+          <Label htmlFor="u-mask">U</Label>
+          <Input
+            id="u-mask"
+            value={uMaskText}
+            onChange={(e) => setUMaskText(e.target.value)}
+          />
+        </div>
+        <div className="flex items-center gap-2">
+          <Label htmlFor="l-mask">L</Label>
+          <Input
+            id="l-mask"
+            value={lMaskText}
+            onChange={(e) => setLMaskText(e.target.value)}
+          />
+        </div>
+        <div className="flex items-center gap-2">
+          <Label htmlFor="f-mask">F</Label>
+          <Input
+            id="f-mask"
+            value={fMaskText}
+            onChange={(e) => setFMaskText(e.target.value)}
+          />
+        </div>
+        <div className="flex items-center gap-2">
+          <Label htmlFor="r-mask">R</Label>
+          <Input
+            id="r-mask"
+            value={rMaskText}
+            onChange={(e) => setRMaskText(e.target.value)}
+          />
+        </div>
+        <div className="flex items-center gap-2">
+          <Label htmlFor="b-mask">B</Label>
+          <Input
+            id="b-mask"
+            value={bMaskText}
+            onChange={(e) => setBMaskText(e.target.value)}
+          />
+        </div>
+        <div className="flex items-center gap-2">
+          <Label htmlFor="d-mask">D</Label>
+          <Input
+            id="d-mask"
+            value={dMaskText}
+            onChange={(e) => setDMaskText(e.target.value)}
+          />
+        </div>
+      </div>
+      <p className="font-bold text-lg">Cases ({cases.length})</p>
+      {cases.map((c, index) => {
+        return (
+          <Card key={index}>
```

Comment:
nit: I think `c.id` can be used as a key instead of index

Reply:
Good catch! Fixed.